### PR TITLE
Python MLC API

### PR DESCRIPTION
**Pull request by bmon581**
Adds support to the MLC backend API.
This adds two main pieces:
1) The ability for the user to select the device. In MLC backend, the user can select device `cpu`, `gpu` or `any`.
2) Introduces the MLC optimizers.
For the MLC device, we strongly encourage users to use the MLC specific devices which would allow us to be more performant on macOS devices.
The existence of the MLC specific optimizers would be temporary as the project remains in prototype. The long term goal would be to properly integrate with the existing PyTorch optimizers.

### CODE REVIEW

**Comment by belmhal** [comment]
will fix the pylint warnings.

File: torch/mlc/mlc_rmsprop.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import torch
+from .mlc_optimizer import MLCOptimizer
+from ..optim.rmsprop import RMSprop
+
+
+class MLCRMSprop(MLCOptimizer):
```
Comment (belmhal) [inline torch/mlc/mlc_rmsprop.py:6]: @smessmer , should the MLC optimizers live on torch.mlc or torch.backends.mlc. Currently, the code is on torch.mlc. but we can move it to another location. Would love to hear your thoughts?

Comment (belmhal) [reply torch/mlc/mlc_rmsprop.py:6]: @smessmer any thoughts on this?

Comment (dlmgary) [reply torch/mlc/mlc_rmsprop.py:6]: @belmhal what's your email? I'd love to collaborate on this.

**Comment by akbir** [comment]
Any update on this?

**Comment by dlmgary** [comment]
Not so much. I'd like to contribute to this issue more but I lack some of the backgrounds to make more updates. @akbir do you understand all of the code that @belmhal has written?

**Comment by Cat7373** [comment]
Why close?

**Comment by mikkelam** [comment]
Facebook is mad at apple over the tracking changes on iPhone 😉

**Comment by matyasbohacek** [comment]
@belmhal Can I ask why this was closed? It seems like this would be an awesome, long-awaited contribution to the Mac data science community

**Comment by rohit901** [comment]
any updates on this?

**Comment by GeekOfComputer01100101** [comment]
> @belmhal Can I ask why this was closed? It seems like this would be an awesome, long-awaited contribution to the Mac data science community

It was closed because MLCompute is not suited for framework backends. See https://github.com/pytorch/pytorch/issues/47702#issuecomment-1033178374 for an explanation.

**Pull request closed without merging**